src/particles/elements/SoftSol.H

```diff
@@ -0,0 +1,439 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_SOFTSOL_H
+#define IMPACTX_SOFTSOL_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "particles/integrators/Integrators.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thick.H"
+
+#include <ablastr/constant.H>
+
+#include <AMReX.H>
+#include <AMReX_Array.H>
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <array>
+#include <cmath>
+#include <tuple>
+
+
+namespace impactx
+{
+    struct Sol_field_data
+    {
+       //   Default:
+       //   Fourier coefficients for the on-axis longitudinal
+       //   magnetic field Bz of a soft-edge solenoid based on
+       //   the design used in the LCLS-II photoinjector.
+       //   Superseded by user-provided input values.
```

Comment:
Reminder for our selves: Let's add a reference (here and in user-facing docs)?

Reply:
A reference now appears here.